File: archive/k/kotlin/Fibonacci.kt
import kotlin.system.exitProcess

// First arg is number of iterations to run
fun main(args: Array<String>) {
    if (args.isNullOrEmpty() || args[0].toIntOrNull()?.takeIf { it >= 0 } == null) {
        println("Usage: please input the count of fibonacci numbers to output")
        return
    }

    val iterations = args[0].toInt()

    var j: Int
    var k = 0
    var l = 1

    for (i in 1..iterations) {
        println("$i: $l")

        j = k
        k = l
        l = j + k
    }
}

File: archive/s/scala/Fibonacci.scala
object Fibonacci:

  private val fibs: LazyList[BigInt] =
    BigInt(1) #:: BigInt(1) #:: fibs.zip(fibs.tail).map(_ + _)
    
  private val usage =
    "Usage: please input the count of fibonacci numbers to output"

  @main def run(args: String*): Unit =
    val output =
      for
        arg <- args.headOption
        n <- arg.toIntOption if n > 0
      yield
        fibs
          .take(n)
          .zipWithIndex
          .map((f, i) => s"${i + 1}: $f")
          .mkString("\n")

    println(output.getOrElse(usage))